File: spring project/animalshome/src/main/java/fmi/course/hcmi/animalshome/dto/PetDto.java
package fmi.course.hcmi.animalshome.dto;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.validation.Valid;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@ToString
@Getter
@Setter
public class PetDto {
    @NotNull
    @Valid
    private PetDetailsDto petDetailsDto;

    @NotNull
    @Valid
    private PetHabitsDto petHabitsDto;

    @NotBlank
    @Size(min = 10, max = 500, message = "The description of the pet must be between 10 and 500 characters long!")
    private String petDescription;
}

File: frontend/Animals-Home/src/app/model/petDto.ts
import { PetDetailsDto } from './petDetailsDto';
import { PetHabitsDto } from './petHabitsDto';

export class PetDto {
    petDetailsDto: PetDetailsDto;
    petHabitsDto: PetHabitsDto;
    petDescription: string;

    public setPetDetails(petDetailsDto: PetDetailsDto){
        this.petDetailsDto = petDetailsDto;
    }

    public setPetHabits(petHabitsDto: PetHabitsDto){
        this.petHabitsDto = petHabitsDto;
    }

    public setPetDescription(petDescription: string){
        this.petDescription = petDescription;
    }
}